pkg/geo/geomfn/validity_check.go

```diff
@@ -64,6 +66,30 @@ func IsValidDetail(g geo.Geometry, flags int) (ValidDetail, error) {
 	}, nil
 }
 
+// IsValidTrajectory returns whether a geometry encodes a valid trajectory
+func IsValidTrajectory(line geo.Geometry) (bool, error) {
+	t, err := line.AsGeomT()
+	if err != nil {
+		return false, err
+	}
+	lineString, ok := t.(*geom.LineString)
+	if !ok {
+		return false, errors.AssertionFailedf("expected linestring, got %T", t)
```

Comment:
we don't want to be surfacing this to the user. (the previous AssertionFailedf comment was because it was impossible to hit). can we use errors.New with `line.ShapeType.String()` instead

Reply:
That makes sense, thanks